### DISCUSSION TITLE
Record on object detection?

### DISCUSSION THREAD

**Discussion by husain3** [Motion-Project/motionplus#8] [Q&A]
When you talk about primary object detection, does that mean that MotionPlus would record when a specific object (i.e. person, car, animal) gets detected? The reason I ask is because my camera has a tree in its view and Motion would record when the wind blows at it.

**Comment by Mr-DaveDev** [comment]
I will say "that is the plan" but there is still some work to do.

The standard deviation threshold parameters are intended to help with spider webs....

The threshold ratio was to help with wind(the number of pixels turning from white to black compared to the count of those going black to white).

And finally the OpenCV secondary detection detects whatever it is that the Haar model is trained to detect.  (A Haar cascade is currently the only secondary detection implemented but other OpenCV models/methods will get added later once the infrastructure is in place).    

The process that I think would occur is that the primary method would trigger and record and then the secondary detection would determine whether to save or delete the images/videos created from the primary detection.  This is needed because the secondary detection through OpenCV consumes a lot of CPU and could not keep up with the frame rate of multiple cameras even on higher powered computers.

**Reply by husain3** [reply]
Where's the code for the object detection stuff that you did? Can I take a look at it?

**Reply by Mr-DaveDev** [reply]
alg_sec.cpp

**Reply by husain3** [reply]
What are your thoughts of having the OpenCV secondary detection just analyze 1-2 frames per second to minimize CPU usage? Do you necessarily have to read every single frame coming from the camera? Or just enough to make a determination of whether or not to save the recorded video? Let me know your thoughts because I haven't experimented with this part of the code yet but have seen raspberrypis successfully do object detection.
